src/main/java/io/micronaut/build/MicronautQualityChecksAggregatorPlugin.java

```diff
@@ -59,22 +54,16 @@ private void configureJacoco(final Project rootProject) {
         ReportingExtension reporting = rootProject.getExtensions().getByType(ReportingExtension.class);
         reporting.getReports().register(COVERAGE_REPORT_TASK_NAME, JacocoCoverageReport.class,
                 r -> r.getTestType().set(TestSuiteType.UNIT_TEST));
-    }
 
-    private void configureJacocoForSubproject(Project subproject) {
-        if (subproject.getPlugins().hasPlugin(JavaPlugin.class)) {
-            subproject.getPluginManager().apply(JacocoPlugin.class);
-        }
-
-        // Do not generate reports for individual projects
-        if (!subproject.getTasks().matching(task -> task.getName().equals(TEST_REPORT_TASK_NAME)).isEmpty()) {
-            subproject.getTasks().named(TEST_REPORT_TASK_NAME).configure(task -> task.setEnabled(false));
-        }
-
-        Project rootProject = subproject.getRootProject();
-        Configuration jacocoAggregation = rootProject.getConfigurations().getByName("jacocoAggregation");
-        jacocoAggregation.getDependencies().add(
-                rootProject.getDependencies().project(Collections.singletonMap("path", subproject.getPath()))
-        );
+        final Configuration jacocoAggregation = rootProject.getConfigurations().getByName("jacocoAggregation");
+        rootProject.getSubprojects().forEach(subproject -> {
+            rootProject.evaluationDependsOn(subproject.getPath());
+            subproject.getPlugins().withType(MicronautQualityChecksParticipantPlugin.class, plugin -> {
+                jacocoAggregation.getDependencies().add(
+                        rootProject.getDependencies().create(subproject)
```

Comment:
:+1: I wish there was a cleaner way to do this with Gradle. This is still fragile but haven't found a better way right now.